subprocess/subtree_oom.go

```diff
@@ -0,0 +1,180 @@
+package subprocess
+
+import (
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type OOMTracker struct {
```

Comment:
does it make sense to make this an interface (and maybe a report type), with the platform specificity in build-flagged functions?

Reply:
It probably makes sense to do this if we intend to add systems in the future. I can rearrange.